lib/puppet/error.rb

```diff
@@ -49,21 +49,46 @@ class ParseErrorWithIssue < Puppet::ParseError
     attr_reader :issue_code, :basic_message, :arguments
     attr_accessor :environment, :node
 
-    # @param message [String] The error message
-    # @param file [String] The path to the file where the error was found
-    # @param line [Integer] The line in the file
-    # @param pos [Integer] The position on the line
-    # @param original [Exception] Original exception
-    # @param issue_code [Symbol] The issue code
-    # @param arguments [Hash{Symbol=>Object}] Issue arguments
+    # From Ruby version 2.1.0 all raised Exceptions know their cause - the exception being processed when
+    # an exception was raised. This conditional initializer does the same for Ruby versions before 2.1.0
     #
-    def initialize(message, file=nil, line=nil, pos=nil, original=nil, issue_code= nil, arguments = nil)
-      super(message, file, line, pos, original)
-      @issue_code = issue_code
-      @basic_message = message
-      @arguments = arguments
-    end
+    if RUBY_VERSION < "2.1.0"
```

Comment:
If we decide to target master instead of 5.5.x then this condition can be removed, as master requires ruby 2.3 or greater.